### PR TITLE
Add checks for invalid keywords to fix #166

### PR DESCRIPTION
**Pull request by ScottPJones**

### CODE REVIEW

File: src/Sundials.jl [hunk 1]
```diff
@@ -5,6 +5,16 @@ module Sundials
 using Compat, DiffEqBase
 import DiffEqBase: solve
 
+const warnkeywords =
+    (:save_idxs, :d_discontinuities, :isoutofdomain, :unstable_check,
+     :calck, :progress, :timeseries_steps, :dense,
```
Comment (ChrisRackauckas) [inline src/Sundials.jl]: no `dense` here anymore.

**Comment by ChrisRackauckas** [comment]
I resolved the merge conflict. Will merge when tests pass.

**Comment by ScottPJones** [comment]
I didn't realized there was a merge conflict, thanks!

**Comment by ChrisRackauckas** [comment]
I caused it.

**Comment by ScottPJones** [comment]
We are stepping all over each other today!

**Pull request merged**